### PR TITLE
Simplify arguments to outdated

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes https://github.com/dart-lang/pub/issues/2393

### CODE REVIEW

File: lib/src/command/outdated.dart [hunk 1]
```diff
@@ -32,11 +32,12 @@ class OutdatedCommand extends PubCommand {
...
-        help: 'Defines how the output should be formatted. Defaults to color '
-            'when connected to a terminal, and no-color otherwise.',
-        valueHelp: 'FORMAT',
-        allowed: ['color', 'no-color', 'json']);
+    argParser.addFlag('color',
+        help: 'Whether to color the output. Defaults to color '
+            'when connected to a terminal, and no-color otherwise.');
+
+    argParser.addFlag('json',
+        help: 'Outputs the results in a json formatted report');
```
Comment (jonasfj) [inline lib/src/command/outdated.dart]: This needs `negatable: false` specifying `--no-json` makes no sense.

File: lib/src/command/outdated.dart [hunk 2]
```diff
@@ -151,11 +152,11 @@ class OutdatedCommand extends PubCommand {
 
     rows.sort();
 
-    if (argResults['format'] == 'json') {
+    if (argResults['json']) {
       await _outputJson(rows);
     } else {
-      final useColors = argResults['format'] == 'color' ||
-          (!argResults.wasParsed('format') && stdin.hasTerminal);
+      final useColors = argResults['color'] ||
```
Comment (jonasfj) [inline lib/src/command/outdated.dart]: Does the default in fact work? Given that `addFlag` is passed `defaultsTo: false` by default. Won't this always be `false` unless called with `--color`?

Can we detect if `--color` was passed? Or if it just assumes the default value?

File: lib/src/command/outdated.dart [hunk 3]
```diff
@@ -32,11 +32,12 @@ class OutdatedCommand extends PubCommand {
...
 
   OutdatedCommand() {
-    argParser.addOption('format',
-        help: 'Defines how the output should be formatted. Defaults to color '
-            'when connected to a terminal, and no-color otherwise.',
-        valueHelp: 'FORMAT',
-        allowed: ['color', 'no-color', 'json']);
+    argParser.addFlag('color',
+        help: 'Whether to color the output. Defaults to color '
+            'when connected to a terminal, and no-color otherwise.');
```
Comment (jonasfj) [inline lib/src/command/outdated.dart]: `defaultsTo: true` might make more sense... depending on how this works.

File: lib/src/command/outdated.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/command/outdated.dart]: Good point.
Done.

File: lib/src/command/outdated.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/command/outdated.dart]: No - wasParsed('color') returns false if there was no explicit color argument.

Comment (sigurdm) [reply lib/src/command/outdated.dart]: Can we detect if --color was passed? Or if it just assumes the default value

wasParsed('color') answers exactly that.

**Pull request merged**